lib/tool/base.rb

```diff
@@ -19,11 +19,18 @@ def self.description(value)
       const_set(:DESCRIPTION, value.tr("\n", " ").strip)
     end
 
-    # Executes the tool and returns the answer
+    # Instantiates and executes the tool and returns the answer
     # @param input [String] input to the tool
     # @return [String] answer
     def self.execute(input:)
-      raise NotImplementedError, "Your tool must implement the `self.execute(input:)` method that returns a string"
+      new.execute(input: input)
+    end
```

Comment:
Tools now work by creating an instance of the tool, but I still keep the class-level method to allow direct execution.